cmd/launcher/svc_config_windows.go

```diff
@@ -103,6 +122,62 @@ func checkDependOnService(launcherServiceKey registry.Key, logger log.Logger) {
 	// Set service to depend on Dnscache
 	serviceList = append(serviceList, dnscacheService)
 	if err := launcherServiceKey.SetStringsValue(dependOnServiceName, serviceList); err != nil {
-		level.Error(logger).Log("msg", "could not set strings value for DependOnService", "err", err)
+		logger.Log(context.TODO(), slog.LevelError,
+			"could not set strings value for DependOnService",
+			"err", err,
+		)
+	}
+}
+
+// checkRestartActions checks the current value of our `SERVICE_FAILURE_ACTIONS_FLAG` and
+// sets it to true if required. See https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_failure_actions_flag
+func checkRestartActions(logger *slog.Logger) {
+	sman, err := mgr.Connect()
+	if err != nil {
+		logger.Log(context.TODO(), slog.LevelError,
+			"connecting to service control manager",
+			"err", err,
+		)
+
+		return
+	}
+
+	defer sman.Disconnect()
+
+	launcherService, err := sman.OpenService(`LauncherKolideK2Svc`)
```

Comment:
Maybe make this a const around where `launcherServiceRegistryKeyName` lives?

Reply:
oh yep! will do